react/Header/Header.less

```diff
@@ -33,6 +27,7 @@
   align-items: center;
   flex: 1 0 auto;
   padding: 0 (@grid-gutter-width / 2);
+  max-width: @grid-container-width + @grid-gutter-width;
```

Comment:
This is sort similar to the footer discussion but because we are only adding only half a gutter padding when there is no outer gutters on page blocks to keep the icons/text off the edge - why we are doing that in the first place I'll have to explore the commit history....